source/slang/slang.cpp

```diff
@@ -1488,6 +1488,14 @@ void Session::addBuiltinSource(
         m_builtinLinkage,
         &sink);
 
+    SourceManager* sourceManager = getBuiltinSourceManager();
+
+    // Set the source manager on the sink
+    sink.sourceManager = sourceManager;
+    // Make the linkage use the builtin source manager
+    Linkage* linkage = compileRequest->getLinkage();
+    linkage->setSourceManager(sourceManager);
```

Comment:
Given that the linkage will be `m_builtinLinkage`, shouldn't this just amount to having the `Session` constructor set the source manager of `m_builtinLinkage` to `getBuiltinSourceManager()`? Doing this on every call to `addBuiltinSource` seems a little odd.

Also: one big thing I might have screwed up in my PR to refactor this stuff is that we could easily have many places where `DiagnosticSink`s are getting constructed without attached source managers. It might be worth it at this point to have `DiagnosticSink` have a real constructor that takes whatever it requires, so that we can make sure all the points where it gets instantiated have been validated.

Reply:
(To be clear, that latter comment is unrelated to this review, and I wouldn't expect a fix for my broken code to get folded into this change)

Reply:
```It might be worth it at this point to have `DiagnosticSink` have a real constructor that takes whatever it requires, so that we can make sure all the points where it gets instantiated have been validated.```

Think this is a good idea.